README.md

```diff
@@ -1,411 +1,492 @@
+```javascript
+const { createReadStream } = require('fs');
+
+const stream = createReadStream('input.txt');
+
+fetch('http://httpbin.org/post', { method: 'post', body: stream })
+    .then(res => res.json())
+    .then(json => console.log(json));
+```
+
+#### Post with form-data (detect multipart)
+```javascript
+const FormData = require('form-data');
+
+const form = new FormData();
+form.append('a', 1);
+
+fetch('http://httpbin.org/post', { method: 'post', body: form })
+    .then(res => res.json())
+    .then(json => console.log(json));
+
+// OR, using custom headers
+// NOTE: getHeaders() is non-standard API
+
+const form = new FormData();
+form.append('a', 1);
+
+const options = {
+    method: 'post',
+    body: form,
+    headers: form.getHeaders()
+}
+
+fetch('http://httpbin.org/post', options)
+    .then(res => res.json())
+    .then(json => console.log(json));
+```
+
+See [test cases](https://github.com/bitinn/node-fetch/blob/master/test/test.js) for more examples.
+
+
+## API
+
+#### fetch(url[, options])
+
+- `url` A string representing the URL for fetching
+- `options` [Options](#fetch-options) for the HTTP(S) request
+- Returns: <code>Promise&lt;[Response](#class-response)&gt;</code>
+
+Perform an HTTP(S) fetch.
+
+`url` should be an absolute url, such as `http://example.com/`. A path-relative URL (`/file/under/root`) or protocol-relative URL (`//can-be-http-or-https.com/`) will result in a rejected promise.
+
+<a id="fetch-options"></a>
+#### Options
+
+The default values are shown after each option key.
+
+```js
+{
+    // These properties are part of the Fetch Standard
+    method: 'GET',
+    headers: {},        // request headers. format is the identical to that accepted by the Headers constructor (see below)
+    body: null,         // request body. can be null, a string, a Buffer, a Blob, or a Node.js Readable stream
+    redirect: 'follow', // set to `manual` to extract redirect headers, `error` to reject redirect
+
+    // The following properties are node-fetch extensions
+    follow: 20,         // maximum redirect count. 0 to not follow redirect
+    timeout: 0,         // req/res timeout in ms, it resets on redirect. 0 to disable (OS limit applies)
+    compress: true,     // support gzip/deflate content encoding. false to disable
+    size: 0,            // maximum response body size in bytes. 0 to disable
+    agent: null         // http(s).Agent instance, allows custom proxy, certificate etc.
+}
+```
+
+##### Default Headers
+
+If no values are set, the following request headers will be sent automatically:
+
+Header            | Value
+----------------- | --------------------------------------------------------
+`Accept-Encoding` | `gzip,deflate` _(when `options.compress === true`)_
+`Accept`          | `*/*`
+`Connection`      | `close` _(when no `options.agent` is present)_
+`Content-Length`  | _(automatically calculated, if possible)_
+`User-Agent`      | `node-fetch/1.0 (+https://github.com/bitinn/node-fetch)`

```

Comment:
Maybe also add `Transfer-Encoding: chunked`? That header is added by Node.js automatically if the input is a stream, I believe.